traveler-ios-ui/TravelerKitUI/Views/LabeledAnnotationView.swift

```diff
@@ -8,7 +8,76 @@
 
 import Foundation
 import MapKit
+import TravelerKit
 
-class LabeledAnnotationView: MKPinAnnotationView {
+open class LabeledAnnotationView: MKAnnotationView {
+    public var pinHorizontalMargin: CGFloat = 6
+    public var pinVerticalMargin: CGFloat = 4
+    public var pinArrowHeight: CGFloat = 6
+    public var pinBorderWidth: CGFloat = 1.5
 
+    public var isChosen: Bool = false {
+        didSet {
+            guard let parkingSpot = annotation as? ParkingSpot else {
+                Log("MapAnnotation not a ParkingSpot", data: nil, level: .error)
+                return
+            }
+
+            image = createAnnotationPinImage(withPrice: parkingSpot.parkingItem.price.valueInBaseCurrency, isChosen: isChosen)
+        }
+    }
+
+    override public func setSelected(_ selected: Bool, animated: Bool) {
+        super.setSelected(selected, animated: animated)
+
+        if selected {
+            isChosen = true
+        }
+    }
+
+    open func createAnnotationPinImage(withPrice price: Double, isChosen: Bool) -> UIImage {
+        let textStyle = NSMutableParagraphStyle()
+        textStyle.alignment = NSTextAlignment.center
+        textStyle.lineBreakMode = NSLineBreakMode.byWordWrapping
+        var textAttributes = [
+            NSAttributedString.Key.font: UIFont.systemFont(ofSize: 13),
+            NSAttributedString.Key.paragraphStyle: textStyle
+        ] as [NSAttributedString.Key : Any]
+
+        let priceText = "$\(Int(price))"
```

Comment:
You can't assume that the currency will always be dollars. What if the base currency in price happens to be euros?. Furthermore, what if somebody else wants to display something else other than price?. What you should do is refactor this annotation to receive a string, and in the place where you're sending the price, you should send the localized description in base currency that, that price has. Also, nowhere in this object should you have a mention of price or any other models.